src/components/ContentSidebar/ContentSidebar.js

```diff
@@ -136,12 +140,15 @@ class ContentSidebar extends PureComponent<Props, State> {
      * @return {void}
      */
     componentDidMount() {
-        const { fileId }: Props = this.props;
+        const { fileId, hasVersions }: Props = this.props;
         this.rootElement = ((document.getElementById(this.id): any): HTMLElement);
         this.appElement = ((this.rootElement.firstElementChild: any): HTMLElement);
 
         if (fileId) {
             this.fetchFile(fileId);
+            if (hasVersions) {
```

Comment:
componentWillReceiveProps also has fetchFile(), this will likely need to be replicated there.
OR maybe have this.fetchVersions() inside fetchFile()

Reply:
Refactoring componentWillReceiveProps a bit